flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/serde/DynamicTableSinkSpecSerdeTest.java

```diff
@@ -173,7 +174,36 @@ static Stream<DynamicTableSinkSpec> testDynamicTableSinkSpecSerde() {
                                         RowType.of(new BigIntType(), new IntType()))),
                         null);
 
-        return Stream.of(spec1, spec2, spec3);
+        Map<String, String> options4 = new HashMap<>();
+        options4.put("connector", TestValuesTableFactory.IDENTIFIER);
+        int[][] targetColumnIndices = new int[][] {{0}, {1}};
+
+        final ResolvedSchema resolvedSchema4 =
+                new ResolvedSchema(
+                        Arrays.asList(
+                                Column.physical("a", DataTypes.BIGINT()),
```

Comment:
Could we have a more complex schema in the test, I suggest nesting with objects and nested arrays, and arrays of objects?
 I am curious if writing nulls to nullable columns is a consideration here; if so maybe a test to ensure that works?

Reply:
@davidradl thx for comments. However, nested columns is not supported by target columns currently. See more in this issue: https://issues.apache.org/jira/browse/FLINK-31301

Reply:
I've added a todo in the comment. Add more complex test cases after FLINK-31301 has been fixed.